### PR TITLE
deprecate the crate client

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: client/src/main/java/io/crate/client/CrateClient.java [hunk 1]
```diff
@@ -54,6 +54,13 @@
 
 import static org.elasticsearch.common.settings.Settings.settingsBuilder;
 
+/**
+ * @deprecated The Crate client and its transport protocol is deprecated
```
Comment (chaudum) [inline client/src/main/java/io/crate/client/CrateClient.java]: ~~is~~ are

File: blackbox/docs/clients/client.txt [hunk 2]
```diff
@@ -7,6 +7,11 @@ Client Usage
 Connect to a Database
 =====================
 
+The Crate client and its transport protocol are deprecated in favor of
```
Comment (chaudum) [inline blackbox/docs/clients/client.txt]: I would put this paragraph into a `.. note:` directive

File: client/src/main/java/io/crate/client/CrateClient.java [hunk 3]
```diff
@@ -54,6 +54,13 @@
 
 import static org.elasticsearch.common.settings.Settings.settingsBuilder;
 
+/**
+ * @deprecated The Crate client and its transport protocol are deprecated
+ *     in favour favor of the PostgreSQL wire protocol.
```
Comment (m-kharbat) [inline client/src/main/java/io/crate/client/CrateClient.java]: typo `in favour favor`

File: blackbox/docs/clients/client.txt [hunk 4]
```diff
@@ -7,6 +7,13 @@ Client Usage
 Connect to a Database
 =====================
 
+.. note::
+
+    The Crate client and its transport protocol are deprecated in favour of the
+    `PostgreSQL wire protocol v3`_. Please use the `PostgreSQL JDBC driver`_
+    instead of the Crate client. For more detailed information on the support of
+    the PostgreSQL wire protocol by Crate, see `Crate Postgres wire protocol`_.
```
Comment (mfussenegger) [inline blackbox/docs/clients/client.txt]: should we already mention that it will be removed in X.Y ?

**Pull request merged**